src/liveEditor/index.ts

```diff
@@ -28,95 +10,64 @@ import {
-import { getFieldType } from "./utils/getFieldType";
-import { isFieldDisabled } from "./utils/isFieldDisabled";
 import liveEditorPostMessage from "./utils/liveEditorPostMessage";
 import { LiveEditorPostMessageEvents } from "./utils/types/postMessage.types";
 
+import initUI from "./components";
+import { addEventListeners, removeEventListeners } from "./listeners";
+import VisualEditorGlobalState from "./globals";
+
 export class VisualEditor {
     private customCursor: HTMLDivElement | null = null;
     private overlayWrapper: HTMLDivElement | null = null;
-    private previousSelectedEditableDOM: Element | null = null;
-    private visualEditorWrapper: HTMLDivElement | null = null;
-    private previousHoveredTargetDOM: Element | null = null;
+    private visualEditorContainer: HTMLDivElement | null = null;
     private focusedToolbar: HTMLDivElement | null = null;
 
     private resizeObserver = new ResizeObserver(([entry]) => {
-        if (!this.overlayWrapper || !this.previousSelectedEditableDOM) return;
-
-        if (!entry.target.isSameNode(this.previousSelectedEditableDOM)) return;
-
-        addFocusOverlay(this.previousSelectedEditableDOM, this.overlayWrapper);
-    });
-
-    private addOverlay(editableElement: Element | null) {
-        if (!this.overlayWrapper || !editableElement) return;
-
-        addFocusOverlay(editableElement, this.overlayWrapper);
-        this.resizeObserver.observe(editableElement);
-    }
-
-    private hideOverlay = (
-        visualEditorOverlayWrapper: HTMLDivElement | null = null
-    ) => {
-        hideFocusOverlay({
-            previousSelectedEditableDOM: this.previousSelectedEditableDOM,
-            visualEditorWrapper: this.visualEditorWrapper,
-            visualEditorOverlayWrapper,
-            focusedToolbar: this.focusedToolbar,
-        });
-
-        if (!this.previousSelectedEditableDOM) return;
-        this.resizeObserver.unobserve(this.previousSelectedEditableDOM);
-        this.previousSelectedEditableDOM = null;
-    };
-
-    private addFocusedToolbar(eventDetails: VisualEditorCslpEventDetails) {
-        const { editableElement } = eventDetails;
-
-        if (!editableElement || !this.focusedToolbar) return;
+        if (
+            !this.overlayWrapper ||
+            !VisualEditorGlobalState.value.previousSelectedEditableDOM
+        )
+            return;
 
-        // Don't append again if already present
         if (
-            this.previousSelectedEditableDOM &&
-            this.previousSelectedEditableDOM === editableElement
-        ) {
+            !entry.target.isSameNode(
+                VisualEditorGlobalState.value.previousSelectedEditableDOM
+            )
+        )
             return;
-        }
 
-        appendFocusedToolbar(eventDetails, this.focusedToolbar);
-    }
+        addFocusOverlay(
+            VisualEditorGlobalState.value.previousSelectedEditableDOM,
+            this.overlayWrapper
+        );
+    });
```

Comment:
Have we tested this code. It looks like the flow should not work

Reply:
The flow seems to work but we still have to write test cases for it.